seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java

```diff
@@ -241,7 +242,15 @@ protected void resolveArchiveCompressedInputStream(
             case GZ:
                 GzipCompressorInputStream gzipIn =
                         new GzipCompressorInputStream(hadoopFileSystemProxy.getInputStream(path));
-                readProcess(path, tableId, output, copyInputStream(gzipIn), partitionsMap, path);
+                GzipParameters parameters = gzipIn.getMetaData();
+                String fileName = parameters.getFilename();
+                if (fileName == null) {
+                    // remove file suffix
+                    // eg: excel need full compressed name
+                    fileName = path.substring(0, path.length() - 3);
+                }
```

Comment:
we should throw exception when excel file name not end with `.gz`

Reply:
get